### PR TITLE
allow initialRouteParams for TabRouters

### PR DESCRIPTION
**Pull request by superburrito**
Allow the passing in of `initialRouteParams` into `TabRouter`'s config since `StackRouter` already has that feature. The request was filed [here](https://github.com/react-navigation/react-navigation/issues/1790).

First PR, please let me know if I've missed something!

### CODE REVIEW

**Comment by brentvatne** [comment]
looks good to me, can you rebase?

**Comment by superburrito** [comment]
@brentvatne Sure 🙂

File: src/routers/TabRouter.js [hunk 1]
```diff
@@ -38,18 +39,21 @@ export default (routeConfigs, config = {}) => {
       let state = inputState;
       if (!state) {
         const routes = order.map(routeName => {
+          const includeInitialParams = routeName === initialRouteName;
```
Comment (satya164) [inline src/routers/TabRouter.js]: Will be cleaner to do `const params = routeName === initialRouteName ? initialRouteParams : undefined;` here and then just use `params` below.

File: src/routers/TabRouter.js [hunk 2]
```diff
@@ -69,6 +73,9 @@ export default (routeConfigs, config = {}) => {
             params: {
               ...route.params,
               ...params,
+              ...(route.routeName === initialRouteName
+                ? initialRouteParams
+                : {}),
```
Comment (satya164) [inline src/routers/TabRouter.js]: Can use `null` here instead of empty object

Comment (superburrito) [reply src/routers/TabRouter.js]: Didn't realise this, thanks!

File: src/routers/TabRouter.js (hunk 1 above)
Comment (superburrito) [reply src/routers/TabRouter.js]: Was unsure about letting the `params` key exist as `params: undefined`... will make the changes.

**Pull request merged**